app/src/main/java/com/example/hris/ui/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.hris.ui.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.hris.repository.HrisRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val hrisRepository: HrisRepository,
+    application: Application
+) : AndroidViewModel(application) {
+
+    val userData = hrisRepository.loginData
```

Comment:
The contents of the `User` object are unused. Maybe just make this `MutableLiveData<Unit>` then set its value when the login succeeds.